lib/nine_one_one/pager_duty_service.rb

```diff
@@ -1,21 +1,22 @@
 module NineOneOne
   class PagerDutyService
     BASE_HOST = 'events.pagerduty.com'.freeze
-    EVENT_ENDPOINT = '/generic/2010-04-15/create_event.json'.freeze
-
+    EVENTS_API_V2_ENDPOINT = '/v2/enqueue'.freeze
     THROTTLE_HTTP_STATUS = 403
     THROTTLE_RETRIES = 2
+    HIGH_URGENCY_ERROR = 'error'.freeze
 
     def initialize(api_integration_key)
       @api_integration_key = api_integration_key
       @http = Http.new(BASE_HOST)
     end
 
-    def trigger_event(incident_key, description, details_hash = nil)
+    def trigger_event(description, source, details_hash = {}, severity = HIGH_URGENCY_ERROR)
```

Comment:
I think that if we are changing the interface, the only required field should be the incident name (description using current naming convention). I'd switch all the others to optional, named parameters. How about that?

Reply:
`PagerDuty#trigger_event`'s parameters changed but it's still [called with old parameters](https://github.com/u2i/nine_one_one/blob/feature/pagerduty_api_v2/lib/nine_one_one.rb#L31).
Is `incident_key` equivalent to `description` and `description` to `source`?

Reply:
Just an idea:
We changed the method signature in a way that is compatible with previous version semantically (it would work with old arguments) but not logically (old args don't have sense in new context).
Maybe we should leave old method, but raise a helpful message when it's used and create a new method for PD v2 support
``` ruby
  def trigger_event(incident_key, description, details_hash = nil)
    raise "#{self.class}#trigger_event is obsolete, ... helpful message how to fix it ..."
  end

  def send_event(description, source, details_hash = {}, severity = HIGH_URGENCY_ERROR)
    # new stuff
  end
```

Reply:
the problem is that e.g. source is also required  - actually only the details_hash is optional

more info about the required params at: https://v2.developer.pagerduty.com/v2/docs/send-an-event-events-api-v2

Reply:
@ppiotrowicz that's what I initially tried to do. I also talked to @mmateja who initially suggested the same but in the end we decided we simply forget about the past and api V1. If one wants to use it then they should not update the gem to the newer version. If one wants, they have to udpate the codebase. We might not like it but it's a common practice.